src/expand.ts

```diff
@@ -161,5 +228,5 @@ export async function expandString(tmpl: string, opts: ExpansionOptions) {
 
   let final_str = tmpl;
   subs.forEach((value, key) => { final_str = replaceAll(final_str, key, value); });
-  return final_str;
+  return { result: final_str, didReplacement: subs.size > 0};
```

Comment:
I think we have a few places where we say to replace "${var}" with "${var}" because it's effectively not supported for that expansion.  Is this going to say those are replaced and "recurse" until the MAX is hit?

Reply:
Recursion is off by default, and so it's not a concern now. However, we can add a check to filter out the case where $*{*} === $*{*}